internal/controller/snapshot/snapshot_adapter_test.go

```diff
@@ -2259,18 +2346,88 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
-		},
-	}
-	err := k8sClient.List(ctx, integrationPipelineRuns, opts...)
-
-	return integrationPipelineRuns.Items, err
-}
+	When("old integration pipelinerun gets cancelled because newer pipelinerun gets triggered", func() {
+
+		It("ensures that old component snapshot together with integration pipelinerun has been canceled", func() {
+			integrationPipelineRunOld.Status = tektonv1.PipelineRunStatus{
+				PipelineRunStatusFields: tektonv1.PipelineRunStatusFields{
+					Results: []tektonv1.PipelineRunResult{},
+				},
+				Status: v1.Status{
+					Conditions: v1.Conditions{
+						apis.Condition{
+							Reason: "Running",
+							Status: "Unknown",
+							Type:   apis.ConditionSucceeded,
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Status().Update(ctx, integrationPipelineRunOld)).Should(Succeed())
+			var buf bytes.Buffer
+			log := helpers.IntegrationLogger{Logger: buflogr.NewWithBuffer(&buf)}
+
+			adapter = NewAdapter(ctx, hasOldCompSnapshotPR, hasApp, log, loader.NewMockLoader(), k8sClient)
+			adapter.context = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ApplicationContextKey,
+					Resource:   hasApp,
+				},
+				{
+					ContextKey: loader.ComponentContextKey,
+					Resource:   hasComp,
+				},
+				{
+					ContextKey: loader.SnapshotContextKey,
+					Resource:   hasOldCompSnapshotPR,
+				},
+				{
+					ContextKey: loader.SnapshotComponentsContextKey,
+					Resource:   []applicationapiv1alpha1.Component{*hasComp},
+				},
+				{
+					ContextKey: loader.AllIntegrationTestScenariosContextKey,
+					Resource:   []v1beta2.IntegrationTestScenario{*integrationTestScenario},
+				},
+				{
+					ContextKey: loader.RequiredIntegrationTestScenariosContextKey,
+					Resource:   []v1beta2.IntegrationTestScenario{*integrationTestScenario},
+				},
+			})
+			//create new group snapshot
+			Expect(k8sClient.Create(adapter.context, hasNewCompSnapshotPR)).Should(Succeed())
+			Eventually(func() error {
+				err := k8sClient.Get(adapter.context, types.NamespacedName{
+					Name:      hasNewCompSnapshotPR.Name,
+					Namespace: "default",
+				}, hasNewCompSnapshotPR)
+				return err
+			}, time.Second*10).ShouldNot(HaveOccurred())
+			result, err := adapter.EnsureIntegrationPipelineRunsExist()
+			Expect(result.CancelRequest).To(BeFalse())
+			Expect(result.RequeueRequest).To(BeFalse())
+			foundStatusCondition := meta.FindStatusCondition(hasOldCompSnapshotPR.Status.Conditions, gitops.AppStudioIntegrationStatusCondition)
+			Expect(foundStatusCondition.Reason).To(Equal(gitops.AppStudioIntegrationStatusCanceled))
+			Expect(err).ToNot(HaveOccurred())
+			statuses, err := gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(hasOldCompSnapshotPR)
+			Expect(err).ToNot(HaveOccurred())
+			detail, ok := statuses.GetScenarioStatus(integrationTestScenario.Name)
+			Expect(ok).To(BeTrue())
+			Expect(detail.Status).To(Equal(intgteststat.IntegrationTestStatusInProgress))
```

Comment:
Curious, why is this integration PLR still in-progress? It belongs to the older cancelled Snapshot, so shouldn't this be in a cancelled state too? I'm sure I'm missing something.

Reply:
What I tried to achieve there, is that I didn't want to affect other integration tests, that are not connected to the snapshot/ component, so this one should not be affected. Hence the status, I can remove this if this makes little sense.

Reply:
that's a good point, I misunderstood between spec.status VS status in snapshot annotation